pcap-linux.c

```diff
@@ -7427,7 +7427,7 @@ fix_offset(pcap_t *handle, struct bpf_insn *p)
 		/*
 		 * What's the offset?
 		 */
-		if (p->k >= SLL2_HDR_LEN) {
+		if ((bpf_int32)p->k >= SLL2_HDR_LEN) {
```

Comment:
Would having an initial check

    	/*
    	 * Existing references to auxiliary data shouldn't be adjusted.
    	 */
    	if (p->k >= SKF_AD_OFF)
    		return 0;

make it clearer what's being tested here?  (It may require a cast to squelch signed vs. unsigned warnings.)

Reply:
That's a good idea.  My version was correct but obscure, this version is correct and clear.